### DISCUSSION TITLE
[Question] How did you test ATS "friendliness"?

### DISCUSSION THREAD

**Discussion by TheColorDarkGreen** [subidit/rover-resume#2] [General]
Hi @subidit, I was hoping you could share how exactly you verified that each of these templates are parsable by applicant tracking systems (ATS). If you used any tools to test/verify, can you please share them? 

Also, why do some templates have the following lines in their .tex files while others don't?

`\pdfgentounicode=1`
`\usepackage[T1]{fontenc}`

**Comment by TheColorDarkGreen** [comment]
bump @subidit

**Comment by faycalki** [comment]
Hi @subidit  I'd like to get an answer to this too. Can you please provide how you checked for ATS?

**Comment by subidit** [comment]
There is no universal ATS used by all recruiters, which makes guaranteeing absolute ATS compatibility challenging. This variability means that no single tool or template can promise 100% compatibility across all systems.

There are some online tools which claims to do ATS check for you but none can provide absolute accuracy due to the aforementioned reason. LinkedIn has a resume upload feature which provides basic parsing feedback. You can also use Python or JavaScript libraries for PDF parsing if you want to go that route. 

But the simplest way to test your resume's ATS-friendliness is surprisingly straightforward. Open your PDF, select all the text, and paste it into a plain text file. This quick test reveals whether your formatting might cause issues. If the text looks garbled, contains weird characters, or loses its structure, it's a sign that your resume might struggle in automated screening processes. Pay attention to how cleanly the text transfers – professional readability is key.

**Comment by subidit** [comment]
> Also, why do some templates have the following lines in their .tex files while others don't?
> 
> `\pdfgentounicode=1` `\usepackage[T1]{fontenc}`

These commands are actually default in most modern LaTeX distributions, so they're typically only required for non-latin character sets.

**Comment by TheColorDarkGreen** [comment]
Hey @subidit @faycalki, here's a solid open-source parser you can test with https://www.open-resume.com/resume-parser created by @xitanggg

https://github.com/xitanggg/open-resume
